SalesforceSDK/Core/Rest/ClientManager.cs

```diff
@@ -78,11 +78,8 @@ public IRestClient PeekRestClient()
                     async () =>
                     {
                         account = AccountManager.GetAccount();
-                        AuthResponse authResponse =
-                            await OAuth2.RefreshAuthTokenRequest(account.GetLoginOptions(), account.RefreshToken);
-                        account.AccessToken = authResponse.AccessToken;
+                        account = await OAuth2.RefreshAuthToken(account);
 
-                        await AuthHelper.PersistCredentialsAsync(account);
```

Comment:
this is done in OAuth2.RefreshAuthToken - this logic was just duplicated in two spots so i consolidated it to RefreshAuthToken